File: src/javatest/designpattern/creational/factory/PcComputer.java
package javatest.designpattern.creational.factory;



public class PcComputer extends Computer {

	private String ram;
	private String hdd;
	private String cpu;
	
	public PcComputer(String ram, String hdd, String cpu){
		this.ram=ram;
		this.hdd=hdd;
		this.cpu=cpu;
	}
	@Override
	public String getRAM() {
		return this.ram;
	}

	@Override
	public String getHDD() {
		return this.hdd;
	}

	@Override
	public String getCPU() {
		return this.cpu;
	}

}

File: src/javatest/designpattern/testdesign/factorytest/PcComputer.kt
package javatest.designpattern.testdesign.factorytest

class PcComputer: Computer() {
    override fun getName(): String {
        return "HP Server"
    }

    override fun getVersion(): String {
        return "2.0"
    }

    override fun getConfiguration(): String {
        return "20 GB Ram 5 TB Hard Disk"
    }
}